### ISSUE TITLE
XBL templates: add `function-available('fr:mode')`

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#5529]
Use case: the `fr:date` component can also be used in plain XForms. However right now it uses `fr:*` functions, which breaks that. To fix this, we'd like to use for example:

```xml
<xf:var
    if-function-available="fr:mode"
    .../>
<xf:var
    if-function-unavailable="fr:mode"
    .../>
```

Attribute names TBD.

**Comment by ebruchez** [comment]
For `fr:date`, we can use these to filter out the `fr:mode()` function. However I note that we have another use an `fr:*` function:

```
xxbl:serialize-external-value="
    DateSupport:serializeExternalValueJava(
        xxf:binding('fr-date'),
        property('xxf:format.input.date'),
        fr:component-param-value('week-start-day')
    )"
```

Here, for `fr:component-param-value('week-start-day')`, we can't use the mechanism suggested above. Maybe we could add support for a child element:

```xml
<xxbl:serialize-external-value
    value="..."
    fr:keep-if-function-available="fr:mode"
/>
```

**Comment by ebruchez** [comment]
Thinking about it, `<xxbl:serialize-external-value>` is not a good solution, because  this goes into `CommonBinding`, and must be constant for the binding. Here you could possibly have conflicting values for `xxbl:serialize-external-value`, and the first one to evaluate would just win, causing unpredictable behavior (for example XForms page breaking Form Runner page or vice-versa).

Could we use instead `saxon:evaluate()` or `xxf:evaluate()` (we expose both)? Something like:

```
xxbl:serialize-external-value="
    DateSupport:serializeExternalValueJava(
        xxf:binding('fr-date'),
        property('xxf:format.input.date'),
        if (...some condition...) then
            xxf:evaluate('fr:component-param-value(''week-start-day'')')
        else
            xxf:evaluate('xxf:component-param-value(''week-start-day'')')
    )"
```

Are there any issues (evaluation context?)? I suppose we need to try, and also to find a condition that tells us whether we are in Form Runner or not.

**Comment by avernet** [comment]
This will also be useful to fix #5533.

**Comment by ebruchez** [comment]
Entered #5568 as follow-up.

- [ ] `fr: keep-if-function-available` -> `xxbl:keep-if-function-available` in `DefaultXblSupport`
- [ ] update doc to mention this is a regression that might be fixed later
